content/en/docs/reference/config-api/kube-proxy-config.v1alpha1.md

```diff
@@ -0,0 +1,537 @@
+---
+title: kube-proxy Configuration (v1alpha1)
+content_type: tool-reference
+package: kubeproxy.config.k8s.io/v1alpha1
+auto_generated: true
+---
+
+
+## Resource Types 
+
+
+  
+- [KubeProxyConfiguration](#kubeproxy-config-k8s-io-v1alpha1-KubeProxyConfiguration)
+  
+    
+
+
+## `KubeProxyConfiguration`     {#kubeproxy-config-k8s-io-v1alpha1-KubeProxyConfiguration}
+    
+
+
+
+
+KubeProxyConfiguration contains everything necessary to configure the
+Kubernetes proxy server.
+
+<table class="table">
+<thead><tr><th width="30%">Field</th><th>Description</th></tr></thead>
+<tbody>
+    
+<tr><td><samp>apiVersion</samp><br/>string</td><td><samp>kubeproxy.config.k8s.io/v1alpha1</samp></td></tr>
+<tr><td><samp>kind</samp><br/>string</td><td><samp>KubeProxyConfiguration</samp></td></tr>
```

Comment:
I think `<samp>` is the wrong element here: it's a fixed string not an example. How about `<tt>`?

Reply:
`<samp>` is a HTML tag added by bootstrap, it is not a "sample".
I tried `tt`, the font size is not the same as the rest of the paragraph, pretty ugly looking in fact. So I didn't use it.

Reply:
Eesh. [`<samp>`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/samp) does represent example text. Maybe `<span class="api-detail">`?

Reply:
@sftim, do you mean we need to add a style sheet for this code highlight? Since the file format is markdown, such a style cannot be added on a per-page basis, we have to revise the site-wide stylesheet.

Reply:
I'm really reluctant to (mis?)use `<samp>` for its rendering side effects. A few very trusting screen readers etc might be fooled there too. So, yes, I'd prefer to add a stylesheet rule - eg to [`_custom.scss`](https://github.com/kubernetes/website/blob/master/assets/scss/_custom.scss).

Reply:
This is becoming ridiculous. `<samp>` is much cleaner a tag than `<tt>` or `<code`. From browser inspect mode, you can see that `<samp>` is affected only by the following two style definitions:

```
pre, code, kbd, samp {
font-family: SFMono-Regular,Menlo,Monaco,Consolas,liberation mono,courier new,monospace;
font-size: 1em;
}

samp {
  font-family: monospace;
}
```

Previously, I tried `<code>` which is more suitable to this context. However, `<code>` style is not very clean. See below style definitions in effect:

```
.td-content p code, .td-content li>code, .td-content table code {
color: inherit;
padding: .2em .4em;
margin: 0;
font-size: 85%;
word-break: normal;
background-color: rgba(0,0,0,.05);
border-radius: .25rem;
}

code {
font-size: 87.5%;
color: #c97300;
word-wrap: break-word;
}
```

You can see that the `<code>` snippet has a smaller font than its surrounding text, and it has background color defined! The background color conflicts with stripped table rows which has a gray background lighter than `<code>`.

I have nothing to say if you insist that 'samp' means 'sample' and we can only use it when the content is a sample. Fine then, we should use 'tt' only when we mean 'teletype', right?

Reply:
To clarify, I said 'samp' is clean because it only applies 'monospace' font family. That is the simplest and cleanest tag for this.

Reply:
> I have nothing to say if you insist that 'samp' means 'sample' 

That is how the web uses it.

In our _current_ theme it's monospace, and in future theme revisions it **might** be too. I've written docs where `<samp>` renders in italic variable-width. I'm worried about the action-at-a-distance effects of assuming `<samp>` always means monospace.

If we define our own class, we can be confident that it's styled how _we_ define it.
BTW `<tt>` is fine to use if Docsy doesn't mind that.

If you can point to the bit of the generator where this is done @tengqm I am willing to make the time to switch it to use a span with a class, and I'm also willing to revise the stylesheet. I do want to make life easy for our future CSS maintainers, and I'm happy to incur the effort of making their lives easier.

Reply:
@sftim I'm against the "inventing our own wheel" approach. The whole website has been using Bootstrap framework for a long time. For the foreseeable future, I don't think we will change that. Inventing a new "class" or "tag" is something we can do, but it is also something we should avoid when possible.

Regarding `<tt>`, I tried it. Its font size is smaller than other text in the same paragraph. 

A much easier way is to switch to `<code>`, if revising the stylesheet is the preferred way. Adding '<span>'s all places will make the page dirty. If you can help fix the `<code>` style by removing its shading background, I can quickly switch the tag back to `<code>` in the generator.

Reply:
The following screenshot gives you an idea why @kbhawkey was challenging the use of `<code>`. It looks weird.

<img width="605" alt="Screen Shot 2021-03-29 at 8 21 43 PM" src="https://user-images.githubusercontent.com/5094756/112835901-783e2980-90cc-11eb-94ea-bc354f2b5988.png">

Reply:
Would `<code class="api-detail">` work? I can CSS that up.

Reply:
Why don't we just remove the background color from the `<code>` tag?

Reply:
Hi. There is built-in styling from bootstrap of the `<samp>` element. The text looks monospacey.
https://getbootstrap.com/docs/5.0/content/reboot/#sample-output

I don't mind the rendered output. The field text and the right hand TOC use the same styled text.

The font/size of text used in the right hand TOC is more difficult to read.
Some of the descriptions with "type/package/path" information use this font; most of the descriptions do not.
The page may look better if all descriptions used the same style.

**Some ideas:**
It might make sense to use the different style (code style) for the `type` of the field, not the descriptions?
You could bold the name of the field and use the common site font for the descriptions and name of field.

The content is not sample computer output, however, the bootstrap `<samp>` style helps distinguish the information in the table. I agree that the site's styling of the `<code>` element could be improved. A custom style for the Configuration API reference pages is a possibility.

It takes some effort to experiment with the rendering of the different elements.
What do you think about agreeing on an HTML element type, add a class, and then fiddle with the styling (no gray background)?
How much effort and time would it take to swap out the `<samp>` element for  a custom `<code class="config-api">` element? If the code is changed, there are several PRs that need to be updated. This takes some time too. I'd like to see these new pages go in for 1.21.